### PR TITLE
 kbfs offline in GUI: redux, saga, and new RPCs

### PR DESCRIPTION
**Pull request by songgao**

### CODE REVIEW

File: go/kbfs/libkbfs/keybase_service_base.go [hunk 1]
```diff
@@ -1154,6 +1154,15 @@ func (k *KeybaseServiceBase) DecryptFavorites(ctx context.Context, dataToEncrypt
 	return k.kbfsClient.DecryptFavorites(ctx, dataToEncrypt)
 }
 
+// NotifyOnlineStatusChanged implements the KeybaseService interface for
+// KeybaseServiceBase.
+func (k *KeybaseServiceBase) NotifyOnlineStatusChanged(ctx context.Context,
+	online bool) error {
+	fmt.Printf("SONGGAO onlienStatus %v\n", online)
```
Comment (songgao) [inline go/kbfs/libkbfs/keybase_service_base.go]: Oops; will remove this when addressing reviews.

**Comment by jakob223** [comment]
On this branch, the `DestinationPicker` story shot fails for me:

```Error: Missing mock react state for 'TopBarSort':```

**Comment by jakob223** [comment]
Also, I rebased on this to use one of its types (I realize I should probably have just held off until it is merged..) and am getting `eslint` complaints about `PathItemInfoOwnProps` being undefined in `common/index.stories.js`.

**Comment by songgao** [comment]
@jakob223 Ah! good catch. I'll fix it now...

File: go/kbfs/simplefs/simplefs.go [hunk 2]
```diff
@@ -2283,7 +2283,15 @@ func (k *SimpleFS) SimpleFSClearConflictState(ctx context.Context,
...
 }
 
-// SimpleFSPing implements the SimpleFSInterface.
-func (k *SimpleFS) SimpleFSPing(ctx context.Context) error {
-	return nil
+// SimpleFSAreWeConnectedToMDServer implements the SimpleFSInterface.
+func (k *SimpleFS) SimpleFSAreWeConnectedToMDServer(ctx context.Context) (bool, error) {
+	// This is kind of expensive, but we are only calling this from GUI right
+	// after KBFS daemon is connected. After that we rely on notifications.
+	// We can change this to diriectly get from currentStatus in KBFSOps if
```
Comment (strib) [inline go/kbfs/simplefs/simplefs.go]: diriectly -> directly

Comment (songgao) [reply go/kbfs/simplefs/simplefs.go]: Oh no the double tap issue of 🍎's butterfly key caps has started to infect Keybase code base.

**Comment by songgao** [comment]
@keybase/react-hackers bump!

**Pull request merged**